src/Security/PasswordValidator.php

```diff
@@ -22,33 +24,73 @@ class PasswordValidator
 {
     use Injectable;
     use Configurable;
+    use Extensible;
 
     /**
      * @config
      * @var array
      */
-    private static $character_strength_tests = array(
+    private static $character_strength_tests = [
         'lowercase' => '/[a-z]/',
         'uppercase' => '/[A-Z]/',
         'digits' => '/[0-9]/',
         'punctuation' => '/[^A-Za-z0-9]/',
-    );
+    ];
 
-    protected $minLength, $minScore, $testNames, $historicalPasswordCount;
+    /**
+     * @config
+     * @var integer
+     */
+    private static $min_length = null;
+
+    /**
+     * @config
+     * @var integer
+     */
+    private static $min_test_score = null;
+
+    /**
+     * @config
+     * @var integer
+     */
+    private static $historic_count = null;
+
+    /**
+     * @var integer
+     */
+    protected $minLength = null;
 
     /**
+     * @var integer
+     */
+    protected $minScore = null;
+
+    /**
+     * @var array
+     */
+    protected $testNames = null;
+
+    /**
+     * @var integer
+     */
+    protected $historicalPasswordCount = null;
+
+    /**
+     * @deprecated 5.0
      * Minimum password length
      *
      * @param int $minLength
      * @return $this
      */
     public function minLength($minLength)
     {
+        Deprecation::notice('5.0', 'Use ->config()->set(\'min_length\', $value) instead.');
```

Comment:
I don't think we should deprecate the instance setters; Remember that config modifications have a cost, and it's entirely possible that you could create multiple instances of PasswordValidator.

Reply:
I'm in favour of deprecating _this one_ though, in favour of `setMinLength()` replacement

Reply:
yeah true, I'm also in favour of deprecating **this** for `setMinLength()` and other more semantic method names.